.changeset/popular-roses-deliver.md

```diff
@@ -0,0 +1,41 @@
+---
+'@makeswift/runtime': patch
+---
+
+Add RichText control.
+
+Until now, having rich text within your custom components required putting a Text component within a Slot control.
+While we are greatful for the flexibility of our primitives we wanted to enable specific component API design.
+The RichText control allows for any number of rich text props, which are then individually editable within our bulder,
+and just like the Slot control, it returns a ReactNode.
+
+```jsx
+function MyHero({ title, body, image, className }) {
+  return (
+    <div className={className}>
+      <div>
+        {title}
+        {body}
+      </div>
+      <Image
+        src={image.url}
+        width={image.dimensions.width}
+        height={image.dimensions.height}
+      />
+    </div>
+  )
+}
+
+ReactRuntime.registerComponent(MyHero, {
+  type: 'my-hero',
+  label: 'My Hero',
+  props: {
+    title: RichText(),
+    body: RichText(),
+    image: Image(),
+    className: Style(),
+  },
+})
+```
+
+Read more about the Richtext control in our [docs](https://www.makeswift.com/docs/controls/rich-text).
```

Comment:
working on this doc. will hold on pushing this till it is ready. this shouldn't be a blocker to approval